packages/core/types/index.js

```diff
@@ -53,6 +62,21 @@ export type ResolvedParcelConfigFile = {|
   resolveFrom?: FilePath,
 |};
 
+export type PreProcessedParcelConfig = {|
```

Comment:
It depends on whether the spread is being processed in ParcelConfig or loadConfig, if it gets processed in loadConfig and no more processing of spread is done in ParcelConfig than it's a Pure Pipeline otherwise it's extendable.

It was mainly to prevent flow errors all over the place.